SCAutolib/controller.py

```diff
@@ -171,11 +199,16 @@ def setup_local_ca(self, force: bool = False):
             raise exceptions.SCAutolibWrongConfig(msg)
 
         ca_dir: Path = self.lib_conf["ca"]["local_ca"]["dir"]
-        self.local_ca = local_ca_factory(ca_dir, force)
+        self.local_ca = local_ca_factory(path=ca_dir, force=force, create=True)
         logger.info(f"Local CA is configured in {ca_dir}")
 
         dump_to_json(self.local_ca)
 
+    def setup_custom_ca(self, card_data: dict):
+        if card_data["card_type"] == "physical":
+            ca = local_ca_factory(create=True, card_data=card_data)
+            dump_to_json(ca)
```

Comment:
it would be nice to have some log messaga if CA wasn't created

Reply:
I do agree, but I have no clear idea of what to check except that ca._ca_cert exists. CustomCA setup is simple and straightforward. If you have an idea for more specific check I will be happy to implement it.